src/shell/shell.cpp

```diff
@@ -61,76 +61,130 @@ DOS_Shell::DOS_Shell() : history(std::make_shared<ShellHistory>())
 	               HELP_Category::Misc,
 	               HELP_CmdType::Program,
 	               "COMMAND"};
-
-	static std::weak_ptr<ShellHistory> global_shell_history = {};
-	history = global_shell_history.lock();
-	if (!history) {
-		history = std::make_shared<ShellHistory>();
-		global_shell_history = history;
-	}
```

Comment:
This is the block I was talking about in my previous comment.  Deleting this breaks the shell history for nested shells.  Test by typing `COMMAND` in the DOS prompt to launch a nested shell.  Type `EXIT` to quit from the nested shell.

I believe what you will observe is that when the nested shell exits and gets destroyed, it will over-write the contents of the shell history text file.  When you quit the root shell, it will over-write that file again and the history of commands you typed in while in the nested shell will be lost.

`ShellHistory` always writes to the same text file which is why I had all `Shell` instances share the same `ShellHistory` object by that static variable.

Reply:
Whoops; definitely didn't mean to delete this chunk! I probably tried something early on and it got dragged in with the 'commit -a' 🤦‍♂️ ; thanks for flagging this @weirddan455 - will get this put back.

Reply:
Ok, all back to normal - thanks @weirddan455.